src/registrar/models/domain.py

```diff
@@ -282,6 +273,184 @@ def _create_host(self, host, addrs):
             logger.error("Error _create_host, code was %s error was %s" % (e.code, e))
             return e.code
 
+    def _convert_list_to_dict(self, listToConvert: list[tuple[str, list]]):
+        """converts a list of hosts into a dictionary
+        Args:
+            list[tuple[str, list]]: such as [("123",["1","2","3"])]
+            This is the list of hosts to convert
+
+        returns:
+            convertDict (dict(str,list))- such as{"123":["1","2","3"]}"""
+        newDict: dict[str, Any] = {}
+
+        for tup in listToConvert:
+            if len(tup) == 1:
+                newDict[tup[0]] = None
+            elif len(tup) == 2:
+                newDict[tup[0]] = tup[1]
+        return newDict
+
+    def isSubdomain(self, nameserver: str):
```

Comment:
is this logic sufficient for determining a subdomain?  or is a regular expression needed?  for example, isSubdomain('ns1-domain.com') would return True for domain with name 'domain.com'.  Is that ok?

Reply:
Great catch! Added regex in 🚀